GUI/GUISlider.cpp

```diff
@@ -511,37 +523,17 @@ GUIPanel *GUISlider::GetPanel(void)
 //////////////////////////////////////////////////////////////////////////////////////////
 // Description:     Calculates the knob position and size.
 
-void GUISlider::CalculateKnob(void)
-{
-    m_KnobPosition = 0;
-    m_KnobSize = 0;
-
-    // Knob image not loaded yet
-    if (!m_KnobImage)
-        return;
-
-    if (m_Orientation == Horizontal) {
-        // Horizontal
-        m_KnobSize = m_KnobImage->GetWidth();
-        
-        if (m_Maximum-m_Minimum > 0) {
-            float V = (float)(m_Value-m_Minimum) / (float)(m_Maximum-m_Minimum);
-            m_KnobPosition = (float)(m_Width-m_KnobSize)*V;
-        }
-
-    } else {
-        // Vertical
-        m_KnobSize = m_KnobImage->GetHeight();
-        
-        if (m_Maximum-m_Minimum > 0) {
-            float V = (float)(m_Value-m_Minimum) / (float)(m_Maximum-m_Minimum);
-            m_KnobPosition = (float)(m_Height-m_KnobSize)*V;
-        }
-    }
-
-    // Clamp the knob position again for the graphics
-    m_KnobPosition = MAX(m_KnobPosition, m_EndThickness);
-    m_KnobPosition = MIN(m_KnobPosition, (m_Orientation == Horizontal ? m_Width : m_Height) - m_KnobSize - m_EndThickness);
+void GUISlider::CalculateKnob(void) {
+	if (!m_KnobImage)
+		return;
+
+	if (m_Maximum > m_Minimum) {
+		const bool orientation = (m_Orientation == Horizontal);
+		m_KnobSize = (orientation) ? m_KnobImage->GetWidth() : m_KnobImage->GetHeight();
+		const int size = (orientation) ? m_Width : m_Height;
+		const float valueRatio = static_cast<float>(m_Value - m_Minimum) / static_cast<float>(m_Maximum - m_Minimum);
+		m_KnobPosition = m_EndThickness + static_cast<int>(static_cast<float>(size - m_KnobSize - 2 * m_EndThickness) * valueRatio);
```

Comment:
In cases like these, I like to put an extra set of brackets around the multiplied terms, just to make order of operations extra clear. Swapping the ordering around so it's m_EndThickness * 2 would probably be a little nicer too, though that may just be a me thing.

Also, where's the 2 come from, is it cause you're dealing with both sides of the slider? And why did you get rid of the `if m_Maximum - m_Minimum > 0 {` check, is there some safety check elsewhere that swaps them if min > max?

Reply:
2 comes from having a visual "end thickness" at both sides, like you said.
The `m_Maximum - m_Minimum > 0` just became `m_Maximum > m_Minimum`, it looks cleaner to me.